### PR TITLE
IR consolidates, tag maintenance and valueMap() implementation

### PR DESCRIPTION
**Pull request by longbinlai**
(1) Implement `valueMap` that allows compiler to express "{a.name, a.age}" to merge multiple-column projection into one single map. 
(2) Move tag maintenance and Auxilia from compiler to IR to make the codes more sound, and add corresponding tests
(3) Unify Alias as NameOrId
(4) Make error information more descriptive
(5) Bug fixes.

### CODE REVIEW

File: research/query_service/ir/runtime/src/process/operator/map/project.rs [hunk 1]
```diff
@@ -106,7 +90,12 @@ impl MapFuncGen for algebra_pb::Project {
                 .expr
                 .ok_or(ParsePbError::from("expr eval is missing in project"))?;
             let evaluator = Evaluator::try_from(expr)?;
-            projected_columns.push((evaluator, alias));
+            let projector = if let Some(prop_key) = evaluator.extract_single_tag() {
```
Comment (longbinlai) [inline research/query_service/ir/runtime/src/process/operator/map/project.rs:93]: 可以直接 `if let Some((tag, prop_key))`, 清楚一些。

Comment (longbinlai) [inline research/query_service/ir/runtime/src/process/operator/map/project.rs:93]: 另外，你这里似乎可以直接从expr判断，如果是single_tag的情况（evaluator里面的extract_single_tag也可以删掉），就不需要建evaluator了。

Comment (BingqingLyu) [reply research/query_service/ir/runtime/src/process/operator/map/project.rs]: Done

Comment (BingqingLyu) [reply research/query_service/ir/runtime/src/process/operator/map/project.rs]: Done

**Pull request merged**